python-lib/dku_timeseries/resampling.py

```diff
@@ -232,6 +237,13 @@ def _fill_in_category_values(self, df, category_columns):
         elif self.params.category_imputation_method == "clip":
             category_filled_df.loc[:, category_columns] = category_filled_df.loc[:, category_columns].ffill().bfill()
         elif self.params.category_imputation_method == "mode":
+            # .mode() loses the timezone info for any datetimetz column
             most_frequent_categoricals = category_filled_df.loc[:, category_columns].mode().iloc[0]
+
+            for col in category_columns:
+                # only perform conversion if the column has a timezone
+                if pd.api.types.is_datetime64_any_dtype(category_filled_df[col]) and category_filled_df[col].dt.tz is not None:
+                    most_frequent_categoricals[col] = most_frequent_categoricals[col].tz_localize("UTC")
```

Comment:
I am concerned that this is maybe dangerous - will expand our integration tests

Reply:
good question - i need to check how mode behaves when we have a not utc timezone - does it auto convert ?

Reply:
So, the plugin doesn't actually support non-utc datatimetz timestamps

```
ValueError: Tz-aware datetime.datetime cannot be converted to datetime64 unless utc=True
```

The fix here is likely easy, but requires more testing to ensure that it's safe, so i think it can wait to see if we have customer demand and we can address it in the future